velox/functions/prestosql/aggregates/tests/PrestoHasherTest.cpp

```diff
@@ -356,4 +358,48 @@ TEST_F(PrestoHasherTest, wrongVectorType) {
   ASSERT_ANY_THROW(hasher.hash(vector, rows, hashes));
 }
 
+TEST_F(PrestoHasherTest, timestampWithTimezone) {
+  const auto toUnixtimeWithTimeZone =
+      [&](const std::vector<std::optional<std::pair<int64_t, std::string>>>&
+              timestampWithTimeZones) {
+        std::vector<std::optional<int64_t>> timestamps;
+        std::vector<std::optional<int16_t>> timeZoneIds;
+        auto size = timestampWithTimeZones.size();
+        timestamps.reserve(size);
+        timeZoneIds.reserve(size);
+
+        for (auto timestampWithTz : timestampWithTimeZones) {
+          if (timestampWithTz.has_value()) {
+            auto timestamp = timestampWithTz.value().first;
+            auto tz = timestampWithTz.value().second;
+            const int16_t tzid =
+                strcmp(tz.c_str(), "+00:00") == 0 ? 0 : util::getTimeZoneID(tz);
```

Comment:
Why do we need to handle +00:00 separately. Any reason util::getTimeZoneID can't process it correctly?

Reply:
It throws a ""Unknown time zone: +00:00" thrown in the test body" for +00:00.

Reply:
Sounds like a bug. Would you take a deeper look to see what's going on? CC: @pedroerp

Reply:
Its because https://github.com/facebookincubator/velox/blob/main/velox/type/tz/TimeZoneDatabase.cpp#L870 doesnt have 00:00 at best it has 00:01 . Also it starts at 1 and not 0. Maybe we ought to add this value there ?

Reply:
Let's discuss with Pedro what's the best way to fix this. CC: @pedroerp

Reply:
That's (sort of) the expected behavior. The problem is that the timezone map is generated based on Presto's timezone database, and that doesn't have a +00:00 entry. We could also extend that library and add a special check for this case, but this would make it less generic.

Reply:
I wonder then how does the following work:

```
presto:di> SELECT timestamp '2012-10-31 01:00 +00:00';
            _col0
-----------------------------
 2012-10-31 01:00:00.000 UTC
```

Reply:
@mbasmanova @pedroerp . Looks like presto explictly doesnt add 0 : https://github.com/prestodb/presto/blob/master/presto-common/src/main/java/com/facebook/presto/common/type/TimeZoneKey.java#L74 and adds it via code : https://github.com/prestodb/presto/blob/master/presto-common/src/main/java/com/facebook/presto/common/type/TimeZoneKey.java#L79

Reply:
@kgpai Nice find, Krishna. Should we do the same?

Reply:
Will create another PR.